core-bundle/src/Intl/Countries.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Intl;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\System;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Intl\Countries as SymfonyCountries;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class Countries
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var ContaoFramework
+     */
+    private $contaoFramework;
+
+    /**
+     * @var array<string>
+     */
+    private $countriesList;
+
+    public function __construct(TranslatorInterface $translator, RequestStack $requestStack, ContaoFramework $contaoFramework, array $countriesList)
+    {
+        $this->translator = $translator;
+        $this->requestStack = $requestStack;
+        $this->contaoFramework = $contaoFramework;
+        $this->countriesList = $countriesList;
+    }
+
+    /**
+     * @return array<string,string> Translated country names indexed by their uppercase ISO 3166-1 alpha-2 code
+     */
+    public function getCountries(string $displayLocale = null): array
+    {
+        if (null === $displayLocale && null !== ($request = $this->requestStack->getCurrentRequest())) {
+            $displayLocale = $request->getLocale();
+        }
+
+        $countries = SymfonyCountries::getNames($displayLocale);
+        $needsResort = false;
+
+        if ($this->countriesList) {
+            $countries = $this->filterCountries($countries, $displayLocale ?? 'en');
+            $needsResort = true;
+        }
+
+        foreach (array_keys($countries) as $countryCode) {
+            $langKey = 'CNT.'.strtolower($countryCode);
+
+            if (
+                $langKey !== ($label = $this->translator->trans($langKey, [], 'contao_countries', $displayLocale))
+                && \is_string($label)
+                && '' !== $label
+            ) {
+                $countries[$countryCode] = $label;
+                $needsResort = true;
+            }
+        }
+
+        if ($needsResort) {
+            (new \Collator($displayLocale ?? 'en'))->asort($countries);
```

Comment:
Shouldn't this use `$GLOBALS['TL_LANGUAGE']` (or `Request::getLocale()`) instead of `en` as fallback?

Reply:
https://github.com/contao/contao/blob/8796a68c24a69e8f7c2e36c0ca205b684fe6d4ba/core-bundle/src/Intl/Countries.php#L57